File: 706-design-hashmap/706-design-hashmap.java
class MyHashMap {
    private List<int[]>[] entries;
    private int size = 13000;
    public MyHashMap() {
        this.entries = new ArrayList[size];
        for(int i=0; i<this.entries.length; i++)
            this.entries[i] = new ArrayList<>();
    }
    
    public int hash(int key){
        return key % this.size;
    }
    public void put(int key, int value) {
        int index = this.hash(key);
        for(int i = 0; i < this.entries[index].size(); i++){
            if(this.entries[index].get(i)[0] == key){
                this.entries[index].get(i)[1] = value;
                return;
            }
        }
        this.entries[index].add(new int[] {key,value});
    }
    
    public int get(int key) {
        int index = this.hash(key);
        for(int i = 0; i < this.entries[index].size(); i++){
            if(this.entries[index].get(i)[0] == key){
                return this.entries[index].get(i)[1];
            }
        }
        return -1;
    }
    
    public void remove(int key) {
        int index = this.hash(key);
        for(int i = 0; i < this.entries[index].size(); i++){
            if(this.entries[index].get(i)[0] == key){
                this.entries[index].remove(i);
                return;
            }
        }
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */

File: 706-design-hashmap/706-design-hashmap.js

var MyHashMap = function() {
    this.array = new Array(13000);
    this.array.fill([]);
};

/** 
 * @param {number} key 
 * @param {number} value
 * @return {void}
 */

let hashing = (key) => {
    return key % 13000;
}

MyHashMap.prototype.put = function(key, value) {
    let hashCode = key % this.array.length;
    for(let i = 0; i < this.array[hashCode].length;i++){
        if(this.array[hashCode][i][0] === key){
            this.array[hashCode][i][1] = value;
            return;
        }
    }
    this.array[hashCode].push([key, value]);
};

/** 
 * @param {number} key
 * @return {number}
 */
MyHashMap.prototype.get = function(key) {
    let hashCode = key % this.array.length;
    for(let i = 0; i < this.array[hashCode].length;i++){
        if(this.array[hashCode][i][0] === key){
            return this.array[hashCode][i][1];
        }
    }
    return -1;
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashMap.prototype.remove = function(key) {
    let hashCode = key % this.array.length;
    for(let i = 0; i < this.array[hashCode].length;i++){
        if(this.array[hashCode][i][0] === key){
            this.array[hashCode].splice(i,1);
        }
    }
    return;
};

/** 
 * Your MyHashMap object will be instantiated and called as such:
 * var obj = new MyHashMap()
 * obj.put(key,value)
 * var param_2 = obj.get(key)
 * obj.remove(key)
 */